pkg/syncer/metadatasyncer.go

```diff
@@ -369,20 +369,41 @@ func pvDeleted(obj interface{}, metadataSyncer *MetadataSyncInformer) {
 		return
 	}
 	var deleteDisk bool
-	if pv.Spec.ClaimRef != nil && (pv.Status.Phase == v1.VolumeAvailable || pv.Status.Phase == v1.VolumeReleased) && pv.Spec.PersistentVolumeReclaimPolicy == v1.PersistentVolumeReclaimDelete {
-		klog.V(3).Infof("PVDeleted: Volume deletion will be handled by Controller")
+	if pv.Spec.PersistentVolumeReclaimPolicy == v1.PersistentVolumeReclaimDelete {
+		if pv.Spec.ClaimRef != nil {
+			if pv.Status.Phase == v1.VolumeAvailable || pv.Status.Phase == v1.VolumeReleased {
+				var metadataList []cnstypes.BaseCnsEntityMetadata
+				pvMetadata := cnsvsphere.GetCnsKubernetesEntityMetaData(pv.Name, nil, true, string(cnstypes.CnsKubernetesEntityTypePV), pv.Namespace)
+				metadataList = append(metadataList, cnstypes.BaseCnsEntityMetadata(pvMetadata))
+
+				updateSpec := &cnstypes.CnsVolumeMetadataUpdateSpec{
+					VolumeId: cnstypes.CnsVolumeId{
+						Id: pv.Spec.CSI.VolumeHandle,
+					},
+					Metadata: cnstypes.CnsVolumeMetadata{
+						ContainerCluster: cnsvsphere.GetContainerCluster(metadataSyncer.cfg.Global.ClusterID, metadataSyncer.cfg.VirtualCenter[metadataSyncer.vcenter.Config.Host].User),
+						EntityMetadata:   metadataList,
+					},
+				}
+				klog.V(4).Infof("PVDeleted: vSphere provisioner removing volume metadata for the pv: %v. Volume deletion will be handled by Controller", pv)
+				if err := volumes.GetManager(metadataSyncer.vcenter).UpdateVolumeMetadata(updateSpec); err != nil {
```

Comment:
Whats the need to delete metadata from the volume if the volume is going to be deleted?